File: big-data-spark/src/main/scala/com/andy/spark/sql/spark1/ParquetLoadData.scala
package com.andy.spark.sql.spark1

import org.apache.spark.sql.SQLContext
import org.apache.spark.{SparkConf, SparkContext}

/**
  * <p>
  *
  * @author leone
  * @since 2019-03-21
  **/
object ParquetLoadData {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("parquet").setMaster("local[*]")

    val sc = new SparkContext(conf)
    val sqlContext = new SQLContext(sc)

    val usersDF = sqlContext.read.parquet("file:///e:/tmp/input/parquet/user.parquet")

    usersDF.createTempView("t_user")

    val dataFrame = sqlContext.sql("select * from t_user where age < 32")

    dataFrame.show()

    dataFrame.rdd.map { row => "name: " + row(1) }.collect().foreach(print)


  }


}

File: big-data-spark/src/main/java/com/andy/spark/sql/ParquetLoadData.java
package com.andy.spark.sql;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SQLContext;

import java.util.List;

/**
 * <p>
 *
 * @author leone
 * @since 2019-03-21
 **/
public class ParquetLoadData {

    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setMaster("local[*]").setAppName("javaSql");
        JavaSparkContext context = new JavaSparkContext(conf);

        SQLContext sqlContext = new SQLContext(context);

        Dataset<Row> parquet = sqlContext.read().parquet("file:///e:/tmp/input/parquet/user.parquet");

        parquet.registerTempTable("t_user");

        sqlContext.sql("select * from t_user where age < 34").show();

        JavaRDD<Row> rowJavaRDD = parquet.javaRDD();

        JavaRDD<String> map = rowJavaRDD.map((Function<Row, String>) row -> "name" + row.getString(1));

        List<String> collect = map.collect();
        System.out.println(collect);


    }

}
